test/integ_tests/test_all_notebooks.py

```diff
@@ -0,0 +1,101 @@
+import json
+import logging
+import os
+import pytest
+
+from testbook import testbook
+from importlib.machinery import SourceFileLoader
+
+# These notebooks have syntax or dependency issues that prevent them from being tested.
+EXCLUDED_NOTEBOOKS = [
+    # These notebooks have cells that have syntax errors
+    "4_Operating_Borealis_beginner_tutorial.ipynb",
+    "bring_your_own_container.ipynb",
+    "qnspsa_with_embedded_simulator.ipynb",
+    # These notebooks have dependency issues
+    "VQE_chemistry_braket.ipynb",
+    "6_Adjoint_gradient_computation.ipynb",
+    # These notebooks are in flux
+    "Using_The_Adjoint_Gradient_Result_Type.ipynb"
+]
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+if "integ_tests" in os.getcwd():
+    os.chdir(os.path.join("..", ".."))
+
+root_path = os.getcwd()
+examples_path = "examples"
+test_notebooks = []
+
+for dir_, _, files in os.walk(examples_path):
+    for file_name in files:
+        if file_name.endswith(".ipynb") and ".ipynb_checkpoints" not in dir_:
+            test_notebooks.append((dir_, file_name))
+
+
+def get_mock_paths(notebook_dir, notebook_file):
+    mock_file = notebook_file.replace(".ipynb", "_mocks.py")
+    split_notebook_dir = notebook_dir.split(os.sep)
+    path_to_root = os.path.join(*([".."] * len(split_notebook_dir)))
+    mock_dir = os.path.join(*split_notebook_dir[1:])
+    path_to_mocks = os.path.join(path_to_root, "test", "integ_tests", mock_dir, mock_file)
+    if not os.path.exists(path_to_mocks):
+        path_to_mocks = os.path.join(path_to_root, "test", "integ_tests", "default_mocks", "default_mocks.py")
+    path_to_utils = os.path.join(path_to_root, "test", "integ_tests", "mock_utils.py")
+    return path_to_utils, path_to_mocks
+
+
+@pytest.mark.parametrize("notebook_dir, notebook_file", test_notebooks)
+def test_all_notebooks(notebook_dir, notebook_file, mock_level):
+    if notebook_file in EXCLUDED_NOTEBOOKS:
+        pytest.skip(f"Skipping Notebook: '{notebook_file}'")
+
+    os.chdir(root_path)
+    os.chdir(notebook_dir)
```

Comment:
I see we are changing the directories at os level in multiple functions, do we need to reset it back once the method execution finishes?

Reply:
We do this to ensure that the notebooks run in their own directory. This is important since some of them reference files with local paths (like helper files/functions). 

Both of the current tests change the working directory, and if we add more they would need to do the same. However, it seems like if we added more they would need to do the same anyway (for the reason specified above).